### DISCUSSION TITLE
How to fix this Auth Failed page React Native

### DISCUSSION THREAD

**Discussion by MrNapcae** [pocketbase/pocketbase#4742] [Q&A]
The provider is configured in the Google console and the Pocketbase admin panel. The first authorization always results in an Auth fail error.
Subsequent ones return successful authorization.

Code: 

 const userData = await pb.collection('users').authWithOAuth2({
        provider: type,
        urlCallback: setUrl,
      });

<WebView
              userAgent="Mozilla/5.0 (Linux; Win64; x64; rv:46.0)r Gecko/20100101 Firefox/68.0"
              onNavigationStateChange={onNavigationStateChange}
              source={{uri: url}}
              style={{flex: 1}}
 />

Logs:
 LOG  webViewState {"webViewState": {"canGoBack": true, "canGoForward": false, "loading": true, "navigationType": "other", "target": 265, "title": "Redirect…", "url": "https://domain.app/api/oauth2-redirect?state=stateCOde&code=4%code&scope=email+profile+openid+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile&authuser=0&prompt=none#"}}
 LOG  webViewState {"webViewState": {"canGoBack": true, "canGoForward": false, "loading": true, "navigationType": "other", "target": 265, "title": "Redirect…", "url": "https://domain.app/_/#/auth/oauth2-redirect-failure"}}

**Comment by MrNapcae** [comment]
Also, from one device, if you try to log in from different emails (I am authorized under 2 emails), then successful authorization will result in constraint failed: UNIQUE constraint failed: _externalAuths.collectionId, _externalAuths.recordId, _externalAuths.provider (2067)

**Comment by ganigeorgiev** [comment]
There are not enough details to understand what the above code snippet does and how your app works.

If you are trying to open The Google OAuth2 sign in page in a webview then I don't think it is possible because Google blocks webviews [since 2021](https://developers.googleblog.com/2021/06/upcoming-security-changes-to-googles-oauth-2.0-authorization-endpoint.html).

See also https://github.com/pocketbase/pocketbase/discussions/2484#discussioncomment-6118501.



> I am authorized under 2 emails...

I don't understand what do you mean by this. One external provider account can have only 1 related PocketBase record within 1 collection.

If you need further help, please create a _minimal_ reproducible repo using for example https://snack.expo.dev.

**Reply by MrNapcae** [reply]
"OAuth2 sign in page in a webview" -  this will work if you specify an agent as in the example above.

I was unable to overcome the error ""EventSource connect took too long." in Error: EventSource connect took too long".
  at https://snack.expo.dev. However, it is not available locally. - https://snack.expo.dev/yXRcrcktEv6YI33TVKIna

Below is the screen that I get after successful authorization in Google. If I repeat the same thing again, everything will go well... I don’t know where to look anymore...

![photo_2024-04-14_16-30-04](https://github.com/pocketbase/pocketbase/assets/42024544/e0d2c3a6-0308-43c5-abea-60444d5ff972)


Repeating the same code I now get:
![123123](https://github.com/pocketbase/pocketbase/assets/42024544/1b1c3079-d010-44a7-8925-bce990a063f4)

**Reply by ganigeorgiev** [reply]
> this will work if you specify an agent as in the example above.

Still I don't recommend doing this.

> I was unable to overcome the error ""EventSource connect took too long." in Error: EventSource connect took too long".

Please read https://github.com/pocketbase/pocketbase/discussions/2484#discussioncomment-6118501.

**Reply by MrNapcae** [reply]
"EventSource connect took too long" -  Locally this error does not exist, because I can configure flipper.

**Reply by ganigeorgiev** [reply]
@MrNapcae I'm confused. Please explain clearly what is the issue and what you are trying todo.

Again, I don't recommend using a webview as this is not guarantee to work and it goes against the Google policies.

**Reply by ganigeorgiev** [reply]
I've just created a test Google OAuth2 app and it is working correctly for me when testing with the following minimal setup - https://snack.expo.dev/_y-uEipvIMWI68vl2gKhF.

If you are still not able to resolve your issue, double check your Google OAuth2 app configuration.
